cmd/ci-images-mirror/main.go

```diff
@@ -322,6 +343,94 @@ func main() {
 	logrus.Info("Process ended gracefully")
 }
 
+func validateConfig(config *quayiociimagesdistributor.CIImagesMirrorConfig, registryConfig string, ciOperatorConfigPath string, configFilePathInReleaseRepo string) error {
+	diff, err := getConfigDiff(configFilePathInReleaseRepo, config)
+	if err != nil {
+		logrus.WithError(err).Fatal("failed to get config diff")
+	}
+	errs := []error{}
+	for _, target := range diff {
+		source := config.SupplementalCIImages[target]
+		if source.Image != "" {
+			if strings.HasPrefix(source.Image, "docker.io/") {
+				errs = append(errs, fmt.Errorf("source image %s is from docker.io", source.Image))
+			}
+			if !isAccessible(source.Image, registryConfig) {
+				errs = append(errs, fmt.Errorf("source image %s is not accessible", source.Image))
+			}
+		}
+		if isPromoted(target, ciOperatorConfigPath) {
+			errs = append(errs, fmt.Errorf("target %s would overwrite a promoted tag", target))
+		}
+		if isAccessible(targetToQuayImage(target), registryConfig) {
+			errs = append(errs, fmt.Errorf("target %s already exists in quay", target))
+		}
+	}
+	return utilerrors.NewAggregate(errs)
+}
+
+func getConfigDiff(configFilePathInReleaseRepo string, config *quayiociimagesdistributor.CIImagesMirrorConfig) ([]string, error) {
+	bytes, err := quayiociimagesdistributor.LoadConfigFromReleaseRepo(configFilePathInReleaseRepo)
+	if err != nil {
+		return []string{}, fmt.Errorf("failed to load config from release repo: %w", err)
+	}
+	masterConfig, err := quayiociimagesdistributor.LoadConfig(bytes)
+	if err != nil {
+		return []string{}, fmt.Errorf("failed to load config file %s: %w", configFilePathInReleaseRepo, err)
+	}
+
+	diff := []string{}
```

Comment:
I would make return type a set of strings which is more friendly with unit testing.